src/groupByKey.ts

```diff
@@ -1,7 +1,21 @@
-type GroupsMap<T> = {
+type GroupsMap <T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const gropedByKey: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const currrentKey = String(item[key]);
+
+    if (gropedByKey[currrentKey]) {
+      return gropedByKey[currrentKey].push(item);
+    }
+
+    gropedByKey[currrentKey] = [];
+
+    return gropedByKey[currrentKey].push(item);
```

Comment:
that's too much, also `forEach` doesn't return anything, invert your solution as I said
```suggestion
    if (!groupedByKey[currentKey]) {
      here add blank array
    }

    here push
```